src/function_run_result.rs

```diff
@@ -28,27 +28,60 @@ pub struct FunctionRunResult {
     pub output: FunctionOutput,
     #[serde(skip)]
     pub profile: Option<String>,
+    #[serde(skip)]
+    pub scale_factor: f64,
+}
+
+const DEFAULT_INSTRUCTIONS_LIMIT: u64 = 11_000_000;
+const DEFAULT_INPUT_SIZE_LIMIT: u64 = 64_000;
+const DEFAULT_OUTPUT_SIZE_LIMIT: u64 = 20_000;
+
+pub fn get_json_size_as_bytes(value: &serde_json::Value) -> usize {
+    serde_json::to_vec(value).map(|v| v.len()).unwrap_or(0)
```

Comment:
I believe we are getting the input and output size in the backend [here](https://github.com/Shopify/shopify/blob/main/gems/functions/lib/functions/execution/instrumentation/event_listeners/publish_runs_to_monorail.rb#L61-L63). And using the `.bytesize`, which I guess is not the exact same. 

But from reading both are measuring the size of data in bytes. But in Ruby, it's the size of a string, and in Rust, it's the size of a serialized JSON structure. Im not sure how to determine if those are similar.

Reply:
The `input_encoded` and `output_encoded` are the minified values. So when we read the byte size of those, we're checking the minified input/output sizes (not the pretty values). These "encoded" strings are the ones that are sent to execute the function with.

Since Function Runner is running on a partner machine, we can't assume that the input will be minified already. However, this made me realize that we don't just want to minify it for the sake of counting the size. We actually want to minify it for running. I think we should ensure that we're minifying the input before execution (probably [here](https://github.com/Shopify/function-runner/blob/main/src/main.rs#L116)).